src/main/java/com/google/sps/servlets/UpdateServlet.java

```diff
@@ -0,0 +1,86 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.ChromeOSDevice;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.gson.Json;
+import com.google.sps.servlets.Util;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import org.json.simple.JSONArray;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/update")
+public class UpdateServlet extends HttpServlet {
+
+  public static final Gson GSON_OBJECT = new Gson();
```

Comment:
Perhaps we can used the centralized Json class for serializing/deserializing instead of creating a new Gson objects every time.  This way we can be sure our custom serializers/deserializers are always used when appropriate.

Reply:
I will make this change; however, I will leave as a note that I do not feel particularly comfortable using our custom serializers/deserializers when not necessarily, given that they have not been rigorously tested with full coverage (and with past errors of our custom serializers/deserializers which this library did not have).